### PR TITLE
Fix submenuItem flicker issue on hover

### PR DESCRIPTION
**Pull request by lenahong**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes
- SubmenuItem.types : add new state submenuItemHovered to keep track of mouse event. Because onHoverIn callback shows the submenu, having focus visuals (background color, text color) depend on the hovered state was causing a flicker.
- SubmenuItem: submenuItemHovered state is flagged true initially onMouseEnter so focus visuals shows up immediately before the submenu flies open with a 500ms delay
- ContextualMenu.types: setsubmenuItemHovered callback is saved in ContextualMenu's context
- ContextualMenuItem (onItemHover): submenuItemHovered state gets set to false as the submenus close on hovering over other menu items.

### Verification

Manually & Automated

### Pull request checklist

This PR has considered (when applicable):
- [X] Automated Tests
- [X] Documentation and examples
- [X] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/ContextualMenu/src/ContextualMenu.types.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ export interface ContextualMenuContext {
    ** ContextualMenuItems will call this submenu dismissal when they are hovered
    */
   dismissSubmenu?: () => void;
+
+  setsubmenuItemHovered?: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (rurikoaraki) [inline packages/components/ContextualMenu/src/ContextualMenu.types.ts:30]: nit: I think the "s" in "Submenu" should be capitalized

Comment (rurikoaraki) [reply packages/components/ContextualMenu/src/ContextualMenu.types.ts:30]: Same with other "setsubmenu..." variables

File: packages/components/ContextualMenu/src/SubmenuItem.types.ts [hunk 2]
```diff
@@ -8,7 +8,10 @@ import { IconProps } from '@fluentui-react-native/icon';
 export const submenuItemName = 'submenuItem';
 export type SubmenuItemTokens = ContextualMenuItemTokens;
 export type SubmenuItemProps = ContextualMenuItemProps;
-export type SubmenuItemState = ContextualMenuItemState;
+
+export interface SubmenuItemState extends ContextualMenuItemState {
+  submenuItemHovered?: boolean;
```
Comment (rurikoaraki) [inline packages/components/ContextualMenu/src/SubmenuItem.types.ts:13]: naming nit: "is" in front of this variable would make it clearer that this is a boolean.

File: packages/components/ContextualMenu/src/SubmenuItem.tsx [hunk 3]
```diff
@@ -38,42 +38,52 @@ export const SubmenuItem = compose<SubmenuItemType>({
...
+      (e) => {
+        setsubmenuItemHovered(true);
+        pressable.props.onMouseEnter && pressable.props.onMouseEnter(e);
+        e.stopPropagation();
+      },
+      [pressable, setsubmenuItemHovered],
+    );
+
     /*
      * For SubmenuItem, menu is shown on hover.
```
Comment (rurikoaraki) [inline packages/components/ContextualMenu/src/SubmenuItem.tsx:84]: nit: might be worth updating this comment to be more precise.

File: packages/components/ContextualMenu/src/SubmenuItem.tsx [hunk 4]
```diff
@@ -38,42 +38,52 @@ export const SubmenuItem = compose<SubmenuItemType>({
...
      */
-    const onKeyUp = useKeyCallback(onItemHoverIn, ' ', 'Enter', 'ArrowRight');
+    const onKeyUp = useKeyCallback(onMouseEnter, ' ', 'Enter', 'ArrowRight');
```
Comment (rurikoaraki) [inline packages/components/ContextualMenu/src/SubmenuItem.tsx:86]: Random question, does mouse enter cover pen interactions? I forget.

**Review by rurikoaraki — APPROVED** [review]
Some nits, but otherwise looks good.

File: packages/components/ContextualMenu/src/SubmenuItem.tsx (hunk 4 above)
Comment (lenahong) [reply packages/components/ContextualMenu/src/SubmenuItem.tsx:86]: I believe so! Pointer events should inherit mouse events

File: packages/components/ContextualMenu/src/SubmenuItem.types.ts (hunk 2 above)
Comment (lenahong) [reply packages/components/ContextualMenu/src/SubmenuItem.types.ts:13]: I think other states omit the "is" i.e. focused, hovered, disabled etc

**Review by ValentinaKozlova — APPROVED** [review]

**Pull request merged**